File: El Paseo/Java/src/main/java/grupo2/model/Producto.java
package grupo2.model;
import java.util.*;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
@Entity
public class Producto {
	@Id @GeneratedValue
    @Column(name="PRODUCTO_ID")
    private Long id;
	private boolean borrado;
	private double precio;
	private String foto;
	private String descripcion;
	private String nombre;
	private int stock;
	@ManyToOne 
	@JoinColumn(name="PRODUCTOR_ID")
	private Productor productor;
	@ManyToOne
	@JoinColumn(name="CATEGORIA_ID")
	private Categoria categoria;
	
	public Producto() {
		super();
	}

	public Long getId() {
		return id;
	}
	
	public boolean isBorrado() {
		return borrado;
	}

	public void setBorrado(boolean borrado) {
		this.borrado = borrado;
	}

	public double getPrecio() {
		return precio;
	}

	public void setPrecio(double precio) {
		this.precio = precio;
	}

	public String getFoto() {
		return foto;
	}

	public void setFoto(String foto) {
		this.foto = foto;
	}

	public String getDescripcion() {
		return descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public int getStock() {
		return stock;
	}

	public void setStock(int stock) {
		this.stock = stock;
	}

	public Productor getProductor() {
		return productor;
	}

	public void setProductor(Productor productor) {
		this.productor = productor;
	}

	public Categoria getCategoria() {
		return categoria;
	}

	public void setCategoria(Categoria categoria) {
		this.categoria = categoria;
	}
	
	
}

File: El Paseo/Angular/src/app/modelos/producto.ts
import { Categoria } from "./categoria";
import { Productor } from "./productor";

export class Producto {
    public id:number;
    public borrado:boolean;
    public precio:number;
    public foto:string;
    public descripcion:string;
    public nombre:string;
    public stock:number;
	public productor:Productor;
	public categoria:Categoria;
    public constructor(id:number,borrado:boolean,precio: number, foto:string, descipcion:string, nombre:string, stock:number,productor:Productor, categoria:Categoria){
        this.id = id;
        this.borrado = borrado;
        this.precio = precio;
        this.foto = foto;
        this.descripcion = descipcion;
        this.nombre = nombre;
        this.stock = stock;
        this.productor = productor;
        this.categoria = categoria;
    }
}
